### ISSUE TITLE
Error rendering site via pkgdown

### ISSUE THREAD

**Issue by orenbenkiki** [r-lib/ragg#61]
TL;DR: `pkgdown` version 1.6.1 fails to find a `ragg_png` function in `ragg` 0.4.0, in R 3.6.2. Did `ragg` change its public API lately, or is this a `pkgdown` bug? Any workaround?

Details:

Working on a new release of my package, I first encountered an issue updating `pkgdown` which boiled down to `textshaped` requiring `fribidi` - I manually installed `fribidi` and could now install `textshaped`, then moved on to reinstalling `ragg` and `pkgdown`. So far, so good, and nothing to do with `ragg` itself.

But now when I render the vignettes I get the following error:

```
...
R -e 'pkgdown::build_site()'

R version 3.6.2 (2019-12-12) -- "Dark and Stormy Night"
Copyright (C) 2019 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> pkgdown::build_site()
-- Installing package into temporary library -----------------------------------
== Building pkgdown site =======================================================
Reading from: '/home/obk/slanter'
Writing to:   '/home/obk/slanter/docs'
-- Initialising site -----------------------------------------------------------
Copying '../R/x86_64-pc-linux-gnu-library/3.6/pkgdown/assets/bootstrap-toc.css' to 'bootstrap-toc.css'
Copying '../R/x86_64-pc-linux-gnu-library/3.6/pkgdown/assets/bootstrap-toc.js' to 'bootstrap-toc.js'
Copying '../R/x86_64-pc-linux-gnu-library/3.6/pkgdown/assets/docsearch.css' to 'docsearch.css'
Copying '../R/x86_64-pc-linux-gnu-library/3.6/pkgdown/assets/docsearch.js' to 'docsearch.js'
Copying '../R/x86_64-pc-linux-gnu-library/3.6/pkgdown/assets/link.svg' to 'link.svg'
Copying '../R/x86_64-pc-linux-gnu-library/3.6/pkgdown/assets/pkgdown.css' to 'pkgdown.css'
Copying '../R/x86_64-pc-linux-gnu-library/3.6/pkgdown/assets/pkgdown.js' to 'pkgdown.js'
Writing '404.html'
-- Building home ---------------------------------------------------------------
Writing 'authors.html'
Writing 'LICENSE-text.html'
-- Building function reference -------------------------------------------------
Writing 'reference/index.html'
Reading 'man/meristems.Rd'
Writing 'reference/meristems.html'
Reading 'man/oclust.Rd'
Writing 'reference/oclust.html'
Reading 'man/reorder_frame.Rd'
Writing 'reference/reorder_frame.html'
Reading 'man/reorder_hclust.Rd'
Writing 'reference/reorder_hclust.html'
Reading 'man/sheatmap.Rd'
Writing 'reference/sheatmap.html'
Reading 'man/slanted_orders.Rd'
Writing 'reference/slanted_orders.html'
Reading 'man/slanted_reorder.Rd'
Writing 'reference/slanted_reorder.html'
-- Building articles -----------------------------------------------------------
Writing 'articles/index.html'
Reading 'vignettes/meristems.Rmd'

-- RMarkdown error -------------------------------------------------------------
Quitting from lines 49-50 (meristems.Rmd)
Error in check_dev(dev) :
  the graphical device'ragg_png'does not exist (as a function)
In addition: Warning messages:
1: 'mode(bg)' differs between new and previous
         ==> NOT changing 'bg'
2: 'mode(bg)' differs between new and previous
         ==> NOT changing 'bg'
3: 'mode(bg)' differs between new and previous
         ==> NOT changing 'bg'
--------------------------------------------------------------------------------
Error : Failed to render RMarkdown

Error: <callr_status_error: callr subprocess failed: Failed to render RMarkdown>
-->
<callr_remote_error: Failed to render RMarkdown>
 in process 17674

 Stack trace:

 Process 17629:
 1. pkgdown::build_site()
 2. pkgdown:::build_site_external(pkg = pkg, examples = examples,  ...
 3. callr::r(function(..., crayon_enabled, crayon_colors, pkgdown_internet) { ...
 4. callr:::get_result(output = out, options)
 5. throw(newerr, parent = remerr[[2]])

 x callr subprocess failed: Failed to render RMarkdown

 Process 17674:
 17. (function (..., crayon_enabled, crayon_colors, pkgdown_internet)  ...
 18. pkgdown::build_site(...)
 19. pkgdown:::build_site_local(pkg = pkg, examples = examples, run_dont_run = r ...
 20. pkgdown:::build_articles(pkg, lazy = lazy, override = override,  ...
 21. purrr::walk(pkg$vignettes$name, build_article, pkg = pkg, quiet = quiet,  ...
 22. purrr:::map(.x, .f, ...)
 23. pkgdown:::.f(.x[[i]], ...)
 24. pkgdown:::render_rmarkdown(pkg, input = input, output = output_file,  ...
 25. base:::tryCatch(callr::r_safe(function(...) rmarkdown::render(...),  ...
 26. base:::tryCatchList(expr, classes, parentenv, handlers)
 27. base:::tryCatchOne(expr, names, parentenv, handlers[[1L]])
 28. value[[3L]](cond)
 29. rlang:::abort("Failed to render RMarkdown", parent = cnd)
 30. rlang:::signal_abort(cnd)
 31. base:::signalCondition(cnd)
 32. (function (e)  ...

 x Failed to render RMarkdown

Execution halted
```

Now, `ragg` itself doesn't appear in the stack traces, but still, `ragg_png`? Needless to say, I don't directly refer to it anywhere in my code base so the issue is probably somewhere in the interface between `pkgdown` and `ragg`.

For reference, here is the log of installing `ragg`, which shows it does use `png15`:

```
Installing package into ‘/home/obk/R/x86_64-pc-linux-gnu-library/3.6’
(as ‘lib’ is unspecified)
trying URL 'https://lib.ugent.be/CRAN/src/contrib/ragg_0.4.0.tar.gz'
Content type 'application/x-gzip' length 414161 bytes (404 KB)
==================================================
downloaded 404 KB

* installing *source* package ‘ragg’ ...
** package ‘ragg’ successfully unpacked and MD5 sums checked
** using staged installation
Found pkg-config cflags and libs!
Using PKG_CFLAGS=-I/usr/include/freetype2 -I/usr/include/libpng15
Using PKG_LIBS=-lfreetype -lpng15 -ltiff   -ljpeg
** libs
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c capture_dev.cpp -o capture_dev.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c init.cpp -o init.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c jpeg_dev.cpp -o jpeg_dev.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c png_dev.cpp -o png_dev.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c ppm_dev.cpp -o ppm_dev.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c tiff_dev.cpp -o tiff_dev.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c agg/src/agg_curves.cpp -o agg/src/agg_curves.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c agg/src/agg_font_freetype.cpp -o agg/src/agg_font_freetype.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c agg/src/agg_trans_affine.cpp -o agg/src/agg_trans_affine.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c agg/src/agg_vcgen_dash.cpp -o agg/src/agg_vcgen_dash.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -I"/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/include" -DNDEBUG  -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/systemfonts/include" -I"/home/obk/R/x86_64-pc-linux-gnu-library/3.6/textshaping/include" -I/usr/local/include -I./agg/include -I/usr/include/freetype2 -I/usr/include/libpng15   -fpic  -g -O2  -c agg/src/agg_vcgen_stroke.cpp -o agg/src/agg_vcgen_stroke.o
ar rcs agg/libstatagg.a agg/src/agg_curves.o agg/src/agg_font_freetype.o agg/src/agg_trans_affine.o agg/src/agg_vcgen_dash.o agg/src/agg_vcgen_stroke.o
/opt/rh/devtoolset-7/root/usr/bin/g++ -std=gnu++11 -shared -L/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/lib -L/usr/local/lib64 -o ragg.so capture_dev.o init.o jpeg_dev.o png_dev.o ppm_dev.o tiff_dev.o -Lagg -lstatagg -lfreetype -lpng15 -ltiff -ljpeg -L/net/mraid14/export/data/users/eladch/tools/CO7/R/3.6.2/lib64/R/lib -lR
installing to /home/obk/R/x86_64-pc-linux-gnu-library/3.6/00LOCK-ragg/00new/ragg/libs
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (ragg)

The downloaded source packages are in
        ‘/tmp/RtmpSi1gB1/downloaded_packages’
```

And it it helps, the log of installing `pkgdown`:

```
Installing package into ‘/home/obk/R/x86_64-pc-linux-gnu-library/3.6’
(as ‘lib’ is unspecified)
trying URL 'https://lib.ugent.be/CRAN/src/contrib/pkgdown_1.6.1.tar.gz'
Content type 'application/x-gzip' length 851091 bytes (831 KB)
==================================================
downloaded 831 KB

* installing *source* package ‘pkgdown’ ...
** package ‘pkgdown’ successfully unpacked and MD5 sums checked
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (pkgdown)

The downloaded source packages are in
        ‘/tmp/RtmpSi1gB1/downloaded_packages’
```

**Comment by thomasp85** [comment]
My guess is you need to update knitr since ragg support is fairly recent there

**Comment by orenbenkiki** [comment]
@thomasp85 : I installed `knitr` version 1.3.0 and it solved the problem. Thanks!
